it/uk/gov/hmrc/excisemovementcontrolsystemapi/repository/ExciseNumberQueueWorkItemRepositoryItSpec.scala

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.excisemovementcontrolsystemapi.repository
+
+import org.bson.types.ObjectId
+import org.mockito.MockitoSugar.when
+import org.scalatest.concurrent.IntegrationPatience
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach, OptionValues}
+import org.scalatestplus.mockito.MockitoSugar.mock
+import org.scalatestplus.play.PlaySpec
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.Application
+import play.api.inject.guice.GuiceApplicationBuilder
+import uk.gov.hmrc.excisemovementcontrolsystemapi.config.AppConfig
+import uk.gov.hmrc.excisemovementcontrolsystemapi.repository.model.ExciseNumberWorkItem
+import uk.gov.hmrc.mongo.TimestampSupport
+import uk.gov.hmrc.mongo.test.{CleanMongoCollectionSupport, PlayMongoRepositorySupport}
+import uk.gov.hmrc.mongo.workitem.ProcessingStatus.{InProgress, ToDo}
+import uk.gov.hmrc.mongo.workitem.WorkItem
+
+import java.time.Instant
+import scala.concurrent.ExecutionContext
+
+class ExciseNumberQueueWorkItemRepositoryItSpec extends PlaySpec
+  with CleanMongoCollectionSupport
+  with PlayMongoRepositorySupport[WorkItem[ExciseNumberWorkItem]]
+  with IntegrationPatience
+  with BeforeAndAfterEach
+  with BeforeAndAfterAll
+  with OptionValues
+  with GuiceOneAppPerSuite {
+
+  protected implicit val ec: ExecutionContext = ExecutionContext.Implicits.global
+  private val appConfig = app.injector.instanceOf[AppConfig]
+  private val dateTimeService = mock[TimestampSupport]
+  private val timestamp = Instant.parse("2018-11-30T18:35:24.00Z")
+
+  protected override val repository = new ExciseNumberQueueWorkItemRepository(
+    appConfig,
+    mongoComponent,
+    dateTimeService
+  )
+
+  protected def appBuilder: GuiceApplicationBuilder =
+    new GuiceApplicationBuilder()
+      .configure(
+        "mongodb.uri" -> mongoUri
+      )
+
+  override implicit lazy val app: Application = appBuilder.build()
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+
+    when(dateTimeService.timestamp()).thenReturn(timestamp)
+  }
+
+  override def afterAll(): Unit = {
+    super.afterAll()
+    dropDatabase()
+  }
+
+  "getWorkItemForErn" should {
+
+    "return Work Item with that ern if one exists in the db " in {
+      val expectedWorkItem = insertWorkItemForErn("ern123")
```

Comment:
Here I would use the MongoSupport functions (insert, etc) as these are very simple. pushNew does extra stuff that we may do not want in tests

Reply:
Updated